### DISCUSSION TITLE
Ingress Redirects & Ingress behavior

### DISCUSSION THREAD

**Discussion by dkraemerwork** [mysticaltech/terraform-hcloud-kube-hetzner#410] [Q&A] [answered]
Hi,

today I setup my first cluster successfully.
Thanks for all your work, this is amazing!

But of course nothing goes smooth the more details you want to change.
I'm a bit confused with how the Ingress is setup in combination with HTTPS redirects.

Currently, my cluster is running and I've setup cert-manager successfully.
When opening our website with HTTPS,  https://apple.xxxx.com an HTTPs connection is established.
But when entering with HTTP, I expect the request to be redirected to HTTPs because of the following config.
this is my described ingress.

_______________________________
Name:             my-ingress
Labels:           <none>
Namespace:        default
Address:          xxx
Ingress Class:    <none>
Default backend:  <default>
TLS:
  xxxx-issuer-account-key terminates apple.xxxx.com
Rules:
  Host               Path  Backends
  apple.xxxx.com  / service:80 (10.42.3.28:80)

Annotations:         cert-manager.io/cluster-issuer: letsencrypt
                     nginx.ingress.kubernetes.io/actions.ssl-redirect:
                       {"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}
                     nginx.ingress.kubernetes.io/force-ssl-redirect: true
                     nginx.ingress.kubernetes.io/from-to-www-redirect: true
                     nginx.ingress.kubernetes.io/listen-ports: [{"HTTP": 80}, {"HTTPS":443}]
                     nginx.ingress.kubernetes.io/rewrite-target: /
                     nginx.ingress.kubernetes.io/ssl-passthrough: true
Events:
  Normal  UpdateCertificate  41m (x6 over 112m)  cert-manager-ingress-shim  Successfully updated Certificate `"xxxx-issuer-account-key"`

___________________________________
Three questions:
- Since the "ingress class" is pointing to none, are all these annotations just not working because this is not an nginx ingress?
- When setting the ingress class name annotation to nginx, it seems like traefik certificates are re-issued. Atleast, the browser is showing "unsecure" because its a self signed traefik certificate. Why does this happen?
- I stumbled upon the  "nginx_ingress_values" field in the kube.tf. If i uncomment that part the whole file is breaking. Is this ment to be just uncommented or how do you use that variable correctly?

My questions might be a bit random, but I can't find any documentation explaining these things.

**Comment by boxedcode** [comment, marked as answer]
The example referenced here solved this challenge for me: https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/239

**Reply by dkraemerwork** [reply]
I will have a look, thank you!

**Reply by mysticaltech** [reply]
@dkraemerwork Yes, the example shared in the issue above is great. Also, you have to research the specific annotations to use depending on which ingress controller you use, by default it's traefik, otherwise it's ingress-nginx.

As a rule of them, for ingress definitions, the helm values variables are not useful, as you have to create them yourself depending on your app and use case. Here is also a great piece of doc to read: https://kubernetes.io/docs/concepts/services-networking/ingress/

As you might have understood by now, this is just part of the kubernetes learning curve and not really related to our specific setup. But always happy to help whenever we can! Keep on reading and testing... You will find your way! :)

And most importantly, do not forget to point your domains of choice, or subdomains to the IP of the LB. You can see this by issuing `kubectl get svc -A`.
